cluster.go

```diff
@@ -889,6 +928,26 @@ func askingMulti(cc conn, ctx context.Context, multi []Completed) *redisresults
 	}
 	results := resultsp.Get(0, len(multi))
 	resps := cc.DoMulti(ctx, commands...)
+	if c.hasLftm {
+		var ml []Completed
+	recover:
+		ml = ml[:0]
+		for i, resp := range resps.s {
+			if resp.Error() == errConnExpired {
+				if i > 0 && commands[i-1] == cmds.AskingCmd {
+					ml = commands[i-1:]
+				} else {
+					ml = commands[i:]
```

Comment:
This reminds me that we need to handle the `MULTI EXEC` pattern correctly: If `errConnExpired` occurs during the `MULTI EXEC`, we should retry the entire `MULTI EXEC` block. This applies to the single client as well, which we overlooked in the previous PR.

Reply:
Would you check this pull-request https://github.com/redis/rueidis/pull/837 . This is draft so I will fix other methods and add tests after checking the implement.

Reply:
@rueian Would you check this pull-request https://github.com/redis/rueidis/pull/837 ? This is draft so I will fix other methods and add tests after checking the implement.

Reply:
Yes, the draft looks good to me.

Reply:
Thanks!

Reply:
@rueian Please review #837 .

Reply:
I'm writing tests for `askingMulti()` but I'm not sure why multi has the following commands when calling that method. Any idea?

```
// Call for unit testing
multi := []Completed{
	client.B().Get().Key("1{t}").Build(),
	client.B().Multi().Build(),
	client.B().Incr().Key("2{t}").Build(),
	client.B().Incr().Key("3{t}").Build(),
	client.B().Exec().Build(),
	client.B().Get().Key("4{t}").Build(),
}
resps := client.DoMulti(context.Background(), multi...)

// multi in askingMulti(). I don't know why there is no [GET 2{t}] [Get 3{t}].
[ASKING]
[GET 1{t}]
[ASKING]
[MULTI]
[EXEC]
[ASKING]
[GET 4{t}]
```

Reply:
Fix https://github.com/redis/rueidis/pull/833/commits/6362824f8c826e8c96313ea41fb92eb51eddf7c6